src/test/java/mate/academy/service/OrderServiceImplTest.java

```diff
@@ -0,0 +1,258 @@
+import mate.academy.repository.orderitem.OrderItemRepository;
+import mate.academy.repository.user.UserRepository;
+import mate.academy.service.impl.OrderServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+
+@ExtendWith(MockitoExtension.class)
+class OrderServiceImplTest {
+    private static final Long USER_ID = 1L;
+    private static final Long ORDER_ID = 1L;
+    @Mock
+    private OrderRepository orderRepository;
+    @Mock
+    private UserRepository userRepository;
+    @Mock
+    private ShoppingCartRepository shoppingCartRepository;
+    @Mock
+    private OrderItemRepository orderItemRepository;
+    @Mock
+    private OrderMapper orderMapper;
+    @Mock
+    private OrderItemMapper orderItemMapper;
+    @InjectMocks
+    private OrderServiceImpl orderService;
+    @Mock
+    private Authentication authentication;
+    private User user;
+    private Book book;
+    private OrderItem orderItem;
+    private Order order;
+    private CartItem cartItem;
+    private OrderItemResponseDto orderItemResponseDto;
+
+    @BeforeEach
+    void setUp() {
+        user = new User()
+                .setId(USER_ID)
+                .setFirstName("John")
+                .setLastName("Doe")
+                .setPassword("test")
+                .setEmail("john@doe.com");
+        book = new Book()
+                .setId(1L)
+                .setIsbn("123456789")
+                .setTitle("Title 1")
+                .setAuthor("Author 1")
+                .setPrice(BigDecimal.valueOf(100))
+                .setDescription("Description 1")
+                .setCoverImage("image1");
+        orderItem = new OrderItem()
+                .setId(1L)
+                .setQuantity(1)
+                .setBook(book)
+                .setPrice(BigDecimal.valueOf(100));
+        order = new Order()
+                .setUser(user)
+                .setOrderDate(LocalDateTime.of(2023, 11, 4, 11,30,1))
+                .setTotal(BigDecimal.valueOf(100))
+                .setStatus(Order.Status.NEW)
+                .setOrderItems(Set.of(orderItem))
+                .setId(1L);
+        cartItem = new CartItem()
+                .setBook(book)
+                .setQuantity(1)
+                .setId(1L);
+        orderItemResponseDto = new OrderItemResponseDto()
+                .setId(1L)
+                .setQuantity(1)
+                .setBookId(1L);
```

Comment:
Such code is better to read, 
and you can use these methods wherever you need
```suggestion
        user = prepareUser();
        book = prepareBook();
        orderItem = prepareOrderItem();
        order = prepareOrder();
        cartItem = prepareCartItem();
        orderItemResponseDto = prepareOrderItemResponseDto();
```